static/js/channel.js

```diff
@@ -31,11 +31,76 @@ Table.renderTable('table', consumerTableOpts, function (tr, item, all) {
   Table.renderCell(tr, 1, queueLink)
   Table.renderCell(tr, 2, ack, 'center')
   Table.renderCell(tr, 3, exclusive, 'center')
-  Table.renderCell(tr, 4, item.prefetch_count, 'right')
+  Table.renderCell(tr, 4, Helpers.formatNumber(item.prefetch_count), 'right')
 })
 
+const prefetchForm = (cb) => {
+  const input = document.createElement('input')
+  input.type = 'number'
+
+  const save = document.createElement('button')
+  save.type = 'submit'
+  save.title = 'Save'
+  save.classList.add('btn-icon', 'btn-submit')
+
+  const reset = document.createElement('button')
+  reset.type = 'reset'
+  reset.title = 'Reset'
+  reset.classList.add('btn-icon', 'btn-reset')
+
+  const form = document.createElement('form')
+  form.classList.add('prefetch-form')
+  form.addEventListener('submit', (event) => {
+    event.preventDefault()
+    const prefetch = parseInt(input.value)
+    HTTP.request('PUT', channelUrl, { body: { prefetch } })
+      .then((r) => {
+        if (!(r && r.is_error)) {
+          cb(prefetch)
+        }
+      })
+  })
+  form.append(input, save, reset)
+  const updateForm = (value) => { input.value = value }
+  return { form, updateForm }
+}
+const prefetchView = (cb) => {
+  const text = document.createElement('div')
+  text.title = 'Click to edit'
+  text.style.cursor = 'pointer'
+  text.addEventListener('click', cb)
+  const updateView = (value) => { text.textContent = Helpers.formatNumber(value) }
+  return { text, updateView }
+}
+
+const prefetchHandler = () => {
```

Comment:
does this have to be a method?

Reply:
no not really, just leftovers from earlier implementations :)

Reply:
But on the other hand, having it in a function means that it doesn't add a bunch of variables on the global scope. 

Alternative would be 


```js
let prefetchHandler
{ // create new scope
  const el = ...
  const update = () => {}
  .... 
  prefetchHandler = { el, update }
}
```

But I find that harder to read